core/src/autogluon/core/searcher/local_searcher.py

```diff
@@ -0,0 +1,135 @@
+import logging
+import pickle
+from collections import OrderedDict
+
+from ..space import Categorical, Space
+
+__all__ = ['LocalSearcher']
+
+logger = logging.getLogger(__name__)
+
+
+class LocalSearcher(object):
+    """Local Searcher (virtual class to inherit from if you are creating a custom Searcher).
+
+    Parameters
+    ----------
+    search_space: dict
+        The configuration space to sample from. It contains the full
+        specification of the Hyperparameters with their priors
+    """
+    def __init__(self, search_space, reward_attribute=None, **kwargs):
+        """
+        :param search_space: Configuration space to sample from or search in
+        :param reward_attribute: Reward attribute passed to update.
+            Default: 'accuracy'
+
+        """
+        self.search_space = search_space
+        self._results = OrderedDict()
+        if reward_attribute is None:
+            reward_attribute = 'accuracy'
+        self._reward_attribute = reward_attribute
```

Comment:
Currently this is to align with API of the old searchers. I'd probably prefer changing this to be some kind of generic `reward` attribute rather than being tied to a particular metric. But will wait to do this until later on as GluonCV may be hard-coded currently to reference this attribute.